src/compiler/crystal/command/playground.cr

```diff
@@ -34,6 +34,10 @@ class Crystal::Command
       end
     end
 
+    playground_url = "http://#{server.host || "127.0.0.1"}:#{server.port || "8080"}"
+
+    system "open #{playground_url}" # Open up the default browser automatically to the Playground.
```

Comment:
Might as well move this out in to a method since you'll probably need to use some `flag` checks on the system anyway. `open_browser(playground_url)` or something like that. Then you can use:

```crystal
def open_browser(playground_url)
  {% if flag?(:darwin) %}
    system "open #{playground_url}"
  {% end %}
end
```

Reply:
That's `xdg-open` on Linux.

Reply:
Great suggestion. Will definitely do that. For these `flag` values, is there a listing of these values I can use? I'm currently looking for a way to tell which OS we're on in order to use the right command.

Reply:
I don't know about a specific place, but there's `:linux`, `:freebsd`, `:openbsd`, `:win32` being used around in the language. I would imagine the bsd ones probably won't need it, and the linux one might need it's own method for checking if `xdg-open` or `sensible-browser`.

Reply:
The supported values for `flags` are essentially components of the target triples. In this case you're interested in the operating system. Supported platforms are listed as subdirectories of `src/lib_c` and on https://github.com/crystal-lang/crystal/wiki/Platform-Support
Currently, that's `linux`, `darwin`, `freebsd`, `openbsd` (and `windows`).

Reply:
This is now a private method called `open_browser` and I added commands for `darwin`, `win32` and `linux` with flag checks. I stubbed out `freebsd` and `openbsd` but I don't know the proper commands to use for those.

I had [trouble building crystal for Ubuntu 18.04](https://github.com/crystal-lang/crystal/issues/5556#issuecomment-469404469) so I wasn't able to test beyond macOS.

Also, I wasn't sure if the windows flag was `win32` or `windows` as @jwoertink said `win32` and @straight-shoota said `windows`.